src/solver/outer/backwardgrad1_outer.cpp

```diff
@@ -0,0 +1,139 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/outer/solvers.hpp>
+
+#include <miopen/outer/invoke_params.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/outer.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/target_properties.hpp>
+
+static const size_t LOCAL_SIZE = 256;
+
+namespace miopen {
+
+namespace solver {
+
+namespace outer {
+
+static bool IsImprovementOverROCm(const miopen::outer::ProblemDescription& problem)
+{
+    auto dtype = problem.GetX1Desc().GetType();
+    auto ydims = problem.GetYDesc().GetLengths();
+    if(ydims[0] <= 32 && ydims[1] <= 128)
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
+}
+
+bool OuterBackwardGrad1::IsApplicable([[maybe_unused]] const ExecutionContext& context,
+                                      const miopen::outer::ProblemDescription& problem) const
+{
+    if(!problem.IsAllPacked())
+        return false;
+    if(!IsImprovementOverROCm(problem))
+        return false;
+    return true;
+}
+
+ConvSolution OuterBackwardGrad1::GetSolution(const ExecutionContext& context,
+                                             const miopen::outer::ProblemDescription& problem) const
+{
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype  = problem.GetX1Desc().GetType();
+    auto x1dims = problem.GetX1Desc().GetLengths();
+    auto x2dims = problem.GetX2Desc().GetLengths();
```

Comment:
Recheck all the unused variables in the solvers.

Reply:
I removed unused variables.